### PR TITLE
Redesign of library to support class inheritance

### PR DESCRIPTION
**Pull request by Chris-Schnaufer**
---
name: Pull Request
about: Fill out this template before submitting a Pull Request.
title: First cut of library
labels: 'pull request'
assignees: ''
---

## Description

First complete cut of the library that has been tested with Transformers.

## Types of changes

Moved code into specialized files and removed some functions that didn't belong in a library (they were moved back to the Transformers that used them)

## Checklist:

- [x] I have read the documentation on [CONTRIBUTING](https://github.com/AgPipeline/Organization-info) 
- [x] My updates require a change to the documentation 
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

## Hints to reviewers:

There is minimal new code here, most of it has been moved from existing Transformers. The biggest changes are related to wrapping some code into classes and moving code out to other files (within the library).

An outstanding issue at this time is being able to publish the library automatically to PyPi when merging into the main branch

### CODE REVIEW

**Comment by jvanderleeuw** [comment]
The library is now able to be automatically published to pypi upon merge to master

**Comment by Chris-Schnaufer** [comment]
> The library is now able to be automatically published to pypi...
It's building and pushing the library to PyPi on push to the main GitHub branch.

File: agpypeline/geoimage.py [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    except Exception as ex:
+        logging.exception("Exception caught while clipping image to plot intersection")
+        raise ex
+
+
+def clip_raster_intersection_json(file_path: str, file_bounds: str, plot_bounds: str, out_file: str) -> Optional[int]:
+    """Clips the raster to the intersection of the file bounds and plot bounds
+    Arguments:
+        file_path: the path to the source file
+        file_bounds: the geometric boundary of the source file as JSON
```
Comment (dlebauer) [inline agpypeline/geoimage.py:100]: is this assuming that each file only has a single polygon?

Comment (Chris-Schnaufer) [reply agpypeline/geoimage.py:100]: Yes, it's assuming that each georeferenced source image only has one polygon that defines its boundaries

**Pull request merged**